### PR TITLE
rename aspnet50/aspnetcore50 TFMs to dnx451/dnxcore50

### PR DESCRIPTION
**Pull request by anurse**
Some of the project.json scripts still need to be updated, that will come eventually but it's mechanical so i don't want to bog down this review.

/cc @davidfowl @muratg @ChengTian @troydai @NTaylorMullen @Eilon

### CODE REVIEW

File: src/Microsoft.Framework.PackageManager/Program.cs [hunk 1]
```diff
@@ -32,6 +33,11 @@ public Program(IServiceProvider hostServices, IApplicationEnvironment environmen
 
         public int Main(string[] args)
         {
+            // Accept and discard the "--debug" switch. The runtime handles it but we
+            // don't want it to confuse us. We also don't want it to appear in help,
+            // so we don't do this via CommandLineApplication
+            args = args.Where(a => a != "--debug").ToArray();
```
Comment (anurse) [inline src/Microsoft.Framework.PackageManager/Program.cs]: I'll kill this before merge.

**Comment by Eilon** [comment]
This looks good to me. That is, every change you made looks good. But I have no way of knowing if this is all :smile:

File: src/Microsoft.Framework.ApplicationHost/Program.cs [hunk 2]
```diff
@@ -29,6 +29,7 @@ public Program(IAssemblyLoaderContainer container, IApplicationEnvironment envir
 
         public Task<int> Main(string[] args)
         {
+            Logger.TraceInformation($"[{nameof(ApplicationHost)}] Launching DNX application on {_environment.RuntimeFramework}");
```
Comment (NTaylorMullen) [inline src/Microsoft.Framework.ApplicationHost/Program.cs]: Should remove, should try n keep eveyrthing as tiny as possible to remove places for possible errors. We can address this in a later issue/pr

File: src/Microsoft.Framework.ApplicationHost/project.json [hunk 3]
```diff
@@ -27,9 +27,9 @@
 
     "scripts": {
         "postbuild": [
-            "%project:Directory%/../../build/batchcopy %project:BuildOutputDir%/Debug/aspnet50/*.* %project:Directory%/../../artifacts/build/dnx-clr-win-x86/bin",
-            "%project:Directory%/../../build/batchcopy %project:BuildOutputDir%/Debug/aspnet50/*.* %project:Directory%/../../artifacts/build/dnx-mono/bin",
-            "%project:Directory%/../../build/batchcopy %project:BuildOutputDir%/Debug/aspnetcore50/*.* %project:Directory%/../../artifacts/build/dnx-coreclr-win-x86/bin"
+            "%project:Directory%/../../build/batchcopy %project:BuildOutputDir%/Debug/aspnet50/*.* %project:Directory%/../../artifacts/build/kre-clr-win-x86/bin",
```
Comment (NTaylorMullen) [inline src/Microsoft.Framework.ApplicationHost/project.json]: You can undo these changes if we want to get my PR in that does all the renames (same below)

File: src/Microsoft.Framework.Runtime.Roslyn.Common/CompilerOptionsExtensions.cs [hunk 4]
```diff
@@ -14,6 +14,7 @@ public static class CompilerOptionsExtensions
     {
         private const string NetFrameworkIdentifier = ".NETFramework";
         private const string AspNetFrameworkIdentifier = "Asp.Net";
+        private const string DnxFrameworkIdentifier = "DNX";
```
Comment (NTaylorMullen) [inline src/Microsoft.Framework.Runtime.Roslyn.Common/CompilerOptionsExtensions.cs]: Why not use the FrameworkNames class?

File: src/Microsoft.Framework.Runtime.Roslyn.Common/project.json [hunk 5]
```diff
@@ -14,5 +14,13 @@
             }
         },
         "aspnetcore50": {}
+    },
+
+    "scripts": {
+        "postbuild": [
```
Comment (NTaylorMullen) [inline src/Microsoft.Framework.Runtime.Roslyn.Common/project.json]: Ahh, will need to run my script on this repo after its in.

**Comment by NTaylorMullen** [comment]
:shipit: once @davidfowl be happy.

File: src/Microsoft.Framework.PackageManager/Program.cs (hunk 1 above)
Comment (davidfowl) [inline src/Microsoft.Framework.PackageManager/Program.cs]: @victorhurdugaci you gotta fix this :smile:

File: src/Microsoft.Framework.Runtime/NuGet/Utility/VersionUtility.cs [hunk 6]
```diff
@@ -80,6 +88,9 @@ public static class VersionUtility
         // These aliases allow us to accept 'wp', 'wp70', 'wp71', 'windows', 'windows8' as valid target farmework folders.
         private static readonly Dictionary<FrameworkName, FrameworkName> _frameworkNameAlias = new Dictionary<FrameworkName, FrameworkName>(FrameworkNameEqualityComparer.Default)
         {
+            { new FrameworkName("Asp.Net, Version=v5.0"), new FrameworkName("DNX, Version=v4.5.1") },
```
Comment (davidfowl) [inline src/Microsoft.Framework.Runtime/NuGet/Utility/VersionUtility.cs]: Not sure about this. What does it do?

File: src/Microsoft.Framework.Runtime/FrameworkReferenceResolver.cs [hunk 7]
```diff
@@ -22,6 +22,10 @@ public class FrameworkReferenceResolver : IFrameworkReferenceResolver
                     new FrameworkName(VersionUtility.NetFrameworkIdentifier, new Version(4, 5, 1))
                 }
             },
+            { new FrameworkName(VersionUtility.DnxFrameworkIdentifier, new Version(4, 5, 1)), new List<FrameworkName> {
```
Comment (davidfowl) [inline src/Microsoft.Framework.Runtime/FrameworkReferenceResolver.cs]: We can do this in a not so hardcoded way since we have 1:1 mappings with desktop now. Instead of hardcoding 4.5.1.

https://github.com/aspnet/XRE/blob/anurse/tfm-rename/src/Microsoft.Framework.Runtime/FrameworkReferenceResolver.cs#L236

File: src/Microsoft.Framework.Runtime/FrameworkReferenceResolver.cs [hunk 8]
```diff
@@ -22,6 +22,10 @@ public class FrameworkReferenceResolver : IFrameworkReferenceResolver
...
+                    new FrameworkName(VersionUtility.NetFrameworkIdentifier, new Version(4, 5, 1))
+                } 
+            }
         };
 
         private static readonly IDictionary<FrameworkName, FrameworkName> _monoAliases = new Dictionary<FrameworkName, FrameworkName>
```
Comment (davidfowl) [inline src/Microsoft.Framework.Runtime/FrameworkReferenceResolver.cs]: You need to update mono aliases

File: src/Microsoft.Framework.Runtime/FrameworkReferenceResolver.cs [hunk 9]
```diff
@@ -85,6 +89,14 @@ public string GetFriendlyFrameworkName(FrameworkName targetFramework)
             {
                 return "ASP.NET 5.0";
             }
+            else if(string.Equals(targetFramework.Identifier, VersionUtility.DnxCoreFrameworkIdentifier, StringComparison.OrdinalIgnoreCase))
```
Comment (davidfowl) [inline src/Microsoft.Framework.Runtime/FrameworkReferenceResolver.cs]: Your if statements not having spaces kills kittens.

File: src/Microsoft.Framework.Runtime/FrameworkReferenceResolver.cs [hunk 10]
```diff
@@ -85,6 +89,14 @@ public string GetFriendlyFrameworkName(FrameworkName targetFramework)
...
+            {
+                return "DNX Core 5.0";
+            }
+            else if(string.Equals(targetFramework.Identifier, VersionUtility.DnxFrameworkIdentifier, StringComparison.OrdinalIgnoreCase))
+            {
+                return "DNX 4.5.1";
```
Comment (davidfowl) [inline src/Microsoft.Framework.Runtime/FrameworkReferenceResolver.cs]: Not quite. This is "DNX "+ targetFramework.Version

File: src/Microsoft.Framework.Runtime.Loader/project.json [hunk 11]
```diff
@@ -24,9 +24,9 @@
 
     "scripts": {
         "postbuild": [
-            "%project:Directory%/../../build/batchcopy %project:BuildOutputDir%/Debug/aspnet50/*.* %project:Directory%/../../artifacts/build/dnx-clr-win-x86/bin",
-            "%project:Directory%/../../build/batchcopy %project:BuildOutputDir%/Debug/aspnet50/*.* %project:Directory%/../../artifacts/build/dnx-mono/bin",
-            "%project:Directory%/../../build/batchcopy %project:BuildOutputDir%/Debug/aspnetcore50/*.* %project:Directory%/../../artifacts/build/dnx-coreclr-win-x86/bin"
+            "%project:Directory%/../../build/batchcopy %project:BuildOutputDir%/Debug/aspnet50/*.* %project:Directory%/../../artifacts/build/kre-clr-win-x86/bin",
```
Comment (davidfowl) [inline src/Microsoft.Framework.Runtime.Loader/project.json]: Whats up with this?

File: src/dnx.host/FrameworkNameUtility.cs [hunk 12]
```diff
@@ -10,13 +11,17 @@ internal class FrameworkNameUtility
     {
         internal static FrameworkName ParseFrameworkName(string frameworkName)
         {
-            if (frameworkName == "aspnet50")
+            if (frameworkName == FrameworkNames.ShortNames.Dnx451)
```
Comment (davidfowl) [inline src/dnx.host/FrameworkNameUtility.cs]: We'll need to expand this eventually.

**Comment by davidfowl** [comment]
I'm hoping that core50 will be added in another PR?

File: src/Microsoft.Framework.Runtime/NuGet/Utility/VersionUtility.cs (hunk 6 above)
Comment (anurse) [inline src/Microsoft.Framework.Runtime/NuGet/Utility/VersionUtility.cs]: It's the same feature that was used to rename "sl4-wp71" to "wp71". We're trying to rename a framework here, so it's the same concept.

Basically all this affects is that BOTH frameworks that come in to IsCompatible go through this dictionary before comparison. Thus, aspnet50 is compared as identical to dnx451, etc.

**Comment by anurse** [comment]
core50 and dnx46 are coming later, since they aren't part of the renames, they are new features.

File: src/Microsoft.Framework.Runtime/FrameworkReferenceResolver.cs (hunk 7 above)
Comment (anurse) [inline src/Microsoft.Framework.Runtime/FrameworkReferenceResolver.cs]: Yes, we can clean it up once everything is working again.

File: src/Microsoft.Framework.ApplicationHost/project.json (hunk 3 above)
Comment (anurse) [inline src/Microsoft.Framework.ApplicationHost/project.json]: Your PR doesn't seem to include this change, but it's only a dev-time thing. I can make a pass over the code this evening to get all of them updated. They aren't required for building though.

**Comment by anurse** [comment]
I think this is as close as I'm going to get it right now. We may as well start pulling it in to dev and see what breaks so we can fix it up. There are some formatting and other minor issues to adjust but nothing that should affect the build. I'll be back online this afternoon.

File: src/Microsoft.Framework.Runtime/FrameworkReferenceResolver.cs (hunk 7 above)
Comment (davidfowl) [inline src/Microsoft.Framework.Runtime/FrameworkReferenceResolver.cs]: File a bug?

File: src/Microsoft.Framework.Runtime/NuGet/Utility/VersionUtility.cs (hunk 6 above)
Comment (davidfowl) [inline src/Microsoft.Framework.Runtime/NuGet/Utility/VersionUtility.cs]: Sure, but what parts of the system get affected? What happens when you kpm build and have both frameworks? What happens if you kpm build and have aspnetcore50? Does it translate into dnxcore50?

**Comment by davidfowl** [comment]
@anurse The mono issue needs to be fixed.

I think the other changes are good, but can you file bugs so we remember to clean this up right after?

I was looking for the other changes we made when we went from k10 to aspnetcore50:

https://github.com/aspnet/XRE/commit/b91f057e531322752bc72f9c0234ee5d7c2bfe40#diff-c4eba54c167015b27bebb155d182b953R104

**Comment by anurse** [comment]
I thought I had the mono fix, but it seems I wasn't quite right :). I'll file issues for the other things now

File: src/Microsoft.Framework.Runtime/NuGet/Utility/VersionUtility.cs (hunk 6 above)
Comment (anurse) [inline src/Microsoft.Framework.Runtime/NuGet/Utility/VersionUtility.cs]: If you kpm build either of aspnetcore50 or dnxcore50, it will only build the section with that exact name. We can clean that up a bit. See #1344

File: src/Microsoft.Framework.Runtime/FrameworkReferenceResolver.cs (hunk 7 above)
Comment (anurse) [inline src/Microsoft.Framework.Runtime/FrameworkReferenceResolver.cs]: #1343

**Comment by anurse** [comment]
#1342 - New TFMs (dnx46 and core50)

File: src/dnx.host/FrameworkNameUtility.cs (hunk 12 above)
Comment (anurse) [inline src/dnx.host/FrameworkNameUtility.cs]: Correct, but as part of #1342 right?
